desci-server/src/controllers/data/move.ts

```diff
@@ -53,79 +44,68 @@ export const moveData = async (req: Request, res: Response<MoveResponse | ErrorR
   }
 
   const latestManifest = await getLatestManifest(uuid, req.query?.g as string, node);
-  const dataBucket = latestManifest?.components?.find((c) => isNodeRoot(c));
 
   try {
+    const newPathSplit = newPath.split('/');
+    const fileName = newPathSplit.pop();
+    const newContextPath = newPathSplit.join('/');
+
     /*
      ** New path collision check
      */
-    const externalCidMap = await generateExternalCidMap(node.uuid);
-    const oldFlatTree = recursiveFlattenTree(await getDirectoryTree(dataBucket.payload.cid, externalCidMap));
-    const hasDuplicates = oldFlatTree.some((oldBranch) => neutralizePath(oldBranch.path) === newPath);
-    if (hasDuplicates) {
+    const noDuplicates = await ensureUniquePathsDraftTree({
+      nodeId: node.id,
+      contextPath: newContextPath,
+      filesBeingAdded: [{ originalname: fileName }],
+    });
+
+    if (!noDuplicates) {
       logger.info('[DATA::Move] Rejected as duplicate paths were found');
       return res.status(400).json({ error: 'Name collision' });
     }
 
-    /*
-     ** Update in dag
+    /**
+     * Update draftNodeTree entries for the move operation
      */
-    const splitContextPath = oldPath.split('/');
-    splitContextPath.shift(); //remove root
-    const fileToMove = splitContextPath.pop();
-    const cleanContextPath = splitContextPath.join('/');
-
-    const splitNewPath = newPath.split('/');
-    splitNewPath.shift(); //remove root
-    const cleanNewPath = splitNewPath.join('/');
-    logger.debug(`[DATA::Move] cleanContextPath: ${cleanContextPath}, Moving: ${fileToMove} to: ${newPath}`);
-    const { updatedDagCidMap, updatedRootCid } = await moveFileInDag(
-      dataBucket.payload.cid,
-      cleanContextPath,
-      fileToMove,
-      cleanNewPath,
-    );
 
-    /*
-     ** Updates old paths in the manifest component payloads to the new ones, updates the data bucket root CID and any DAG CIDs changed along the way
-     */
-    let updatedManifest = updateComponentPathsInManifest({
-      manifest: latestManifest,
-      oldPath: oldPath,
-      newPath: newPath,
+    const entriesToUpdate = await prisma.draftNodeTree.findMany({
+      where: {
+        nodeId: node.id,
+        OR: [
+          {
+            path: {
+              startsWith: oldPath + '/',
+            },
+          },
+          {
+            path: oldPath,
+          },
+        ],
+      },
```

Comment:
Re. the previous comment about table structure. This would just be removing the child from one dir, and adding it as a child to another. All subpaths would automatically be correct as they know nothing about their heritage. Still not sure that I'd work, but wanted to share some more context :)